perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java

```diff
@@ -145,10 +161,23 @@ public AuthzRoles call(ApiCaller ac, Deserializer parms) throws PerunException {
 				throw new RpcException(RpcException.Type.WRONG_PARAMETER, "Role with name " + roleName + " does not exist.");
 			}
 			roleName = roleName.toUpperCase();
+			int complementaryObjectId = parms.readInt("complementaryObjectId");
+			String complementaryObjectName = parms.readString("complementaryObjectName");
+
+			PerunBean bean = null;
+			try {
+				bean = (PerunBean) Class.forName("cz.metacentrum.perun.core.api." + complementaryObjectName).getConstructor().newInstance();
+				bean.setId(complementaryObjectId);
+			} catch (InstantiationException e) {
+				e.printStackTrace();
```

Comment:
This is probably not the right way how to process an exception in RPC.

Reply:
Fixed